index expression in binary map

The following program `sample.go` triggers a panic:

```go
package main

import "net/http"
    
type T struct {
    header string
}
    
func (b *T) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
    if b.header != "" {
        req.Header[b.header] = []string{"hello"}
    }
}
    
func main() {
    println("ok")
}
```

Expected result:
```console
$ go run ./sample.go
ok
```

Got:
```console
$ yaegi ./sample.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x12095da]

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*Type).TypeOf(0x0, 0xc0000a07b0, 0x17fb080)
	/Users/marc/go/src/github.com/containous/yaegi/interp/type.go:625 +0x3a
github.com/containous/yaegi/interp.getIndexMap(0xc0001b6c00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/run.go:923 +0xb0
github.com/containous/yaegi/interp.setExec.func1(0xc0001b6c00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1578 +0xb2
github.com/containous/yaegi/interp.setExec.func1(0xc0001b7000)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1567 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0001b6d00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1567 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0001b6500)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1567 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0001b6600)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1567 +0x25d
github.com/containous/yaegi/interp.setExec(0xc0001b6600)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1581 +0xf5
github.com/containous/yaegi/interp.genRun.func1(0xc0000e7600, 0xc00018dd01)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1378 +0x87
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e7600, 0xc00018ddb8, 0x0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:106 +0xc9
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e6f00, 0xc00018ddb8, 0x0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e6200, 0xc00018ddb8, 0x0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.genRun(0xc0000e6200, 0xc000143ec0, 0xc0000a0680)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1370 +0x64
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000082540, 0xc0001b2000, 0xeb, 0x2eb, 0xc0001b2000, 0xeb, 0x0, 0xc000195040)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:276 +0x2a1
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:61 +0x443
```

--- a/_test/map17.go
+++ b/_test/map17.go
@@ -1,0 +1,20 @@
+package main
+
+import "net/http"
+
+type T struct {
+	header string
+}
+
+func (b *T) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	if b.header != "" {
+		req.Header[b.header] = []string{"hello"}
+	}
+}
+
+func main() {
+	println("ok")
+}
+
+// Output:
+// ok

--- a/interp/run.go
+++ b/interp/run.go
@@ -917,7 +917,7 @@ func getIndexArray(n *Node) {
 func getIndexMap(n *Node) {
 	value0 := genValue(n.child[0]) // map
 	tnext := getExec(n.tnext)
-	z := reflect.New(n.child[0].typ.val.TypeOf()).Elem()
+	z := reflect.New(n.child[0].typ.TypeOf().Elem()).Elem()
 
 	if n.child[1].rval.IsValid() { // constant map index
 		mi := n.child[1].rval

